src/chapters/setup-intro-data-types/more-data-types.rst

```diff
@@ -2,258 +2,321 @@ More Data Types
+will adopt this terminology from now on.
 
-   public static void main(String[] args)
-   {
-       // some code
-   }
+.. admonition:: Warning
 
-We’ve already explored each element of this line, however, we haven’t
-really shown you how you might create your own methods in other
-contexts. To do so involves using a different name for our method, and
-adjusting the return type and parameter types accordingly.
+   Be prepared for push-back and negative judgement from Java coders if you
+   accidentally refer to methods as “functions”.
 
-.. raw:: html
+.. index:: ! static methods
 
-   <aside class="aside-warning">
+We’ll dive deeper into classes and objects in Java soon enough. For now, we
+will explore how to write methods. In particular, we’ll use **static methods**.
+A static method is one that can be called without creating an instance of the
+class to which it belongs.
 
-Until we get into Object Oriented Programming, every method we write
-will have the ``static`` keyword. Leaving off ``static`` will prevent
-you from calling methods you define as you would like to.
+.. admonition:: Example
 
-We will explore exactly what ``static`` does in more detail in later
-lessons.
+   Define the class ``Cat`` and include the ``static`` keyword before the
+   ``makeNoise`` method name:
 
-.. raw:: html
+   .. sourcecode:: java
 
-   </aside>
+      public class Cat {
+         public static void makeNoise(String[] args) {
+            // some code
+         }
+      }
 
-Let’s create two classes in Java to demonstrate this. One will have a
-``main`` method and the other will have a method that we want to call
-from within ``main``.
+   Since ``makeNoise`` is ``static``, we do NOT need to create a ``Cat`` object to
+   access it.
 
-.. code:: java
+   Instead of doing this:
 
-   public class HelloMethods {
+   .. sourcecode:: java
+      :linenos:
 
-       public static void main(String[] args) {
-           String message = Message.getMessage("fr");
-           System.out.println(message);
-       }
+      Cat myCat = new Cat();     // Create a new Cat object.
+      myCat.makeNoise("purr");   // Call the makeNoise method.
 
-   }
+   We can call the method directly:
 
-.. code:: java
+   .. sourcecode:: java
+      :linenos:
 
-   public class Message {
+      Cat.makeNoise("roar");
 
-       public static String getMessage(String lang) {
+Until we get further into Object Oriented Programming, every method you write
+should use the ``static`` keyword. Leaving off ``static`` will prevent or
+complicate the process of calling the methods you defined.
 
-           if (lang.equals("sp")) {
-               return "Hola Mundo";
-           } else if (lang.equals("fr")) {
-               return "Bonjour le monde";
-           } else {
-               return "Hello World";
-           }
-       }
-   }
+We will explore exactly what ``static`` does in more detail in later lessons.
+
+Static Method Examples
```

Comment:
for these examples, can you add them to the java-web-dev-exercises repo so the students can play along with the code?

Reply:
Done, but I have a question about ``package org.launchcode.java.demos.javawebdevelopment;``.